packages/back-end/src/controllers/features.ts

```diff
@@ -1172,20 +1177,110 @@ export async function postFeatureRule(
+    }
+    if (interfaceFields.datasourceId === undefined) {
+      throw new Error("Datasource is required for safe rollouts");
+    }
+    if (interfaceFields.guardrailMetricIds === undefined) {
+      throw new Error("Guardrail metrics are required for safe rollouts");
+    }
+
+    const metricIds = interfaceFields.guardrailMetricIds;
+    const datasourceId = interfaceFields.datasourceId;
+    if (metricIds.length) {
+      const map = await getMetricMap(context);
+      for (let i = 0; i < metricIds.length; i++) {
+        const metric = map.get(metricIds[i]);
+        if (metric) {
+          if (datasourceId && metric.datasource !== datasourceId) {
+            throw new Error(
+              "Metrics must be tied to the same datasource as the safe rollout: " +
+                metricIds[i]
+            );
+          }
+        } else {
+          // check to see if this metric is actually a metric group
+          const metricGroup = await context.models.metricGroups.getById(
+            metricIds[i]
+          );
+          if (metricGroup) {
+            // Make sure it is tied to the same datasource as the experiment
+            if (datasourceId && metricGroup.datasource !== datasourceId) {
+              throw new Error(
+                "Metrics must be tied to the same datasource as the safe rollout: " +
+                  metricIds[i]
+              );
+            }
+          } else {
+            // new metric that's not recognized...
+            throw new Error("Invalid metric specified: " + metricIds[i]);
+          }
+        }
+      }
+    }
+    rule.status = "running";
+    rule.seed = rule.seed || uuidv4();
+    rule.trackingKey = rule.trackingKey || `srk_${uuidv4()}`;
+    const safeRolloutCreateProps: CreateProps<SafeRolloutInterface> = {
+      ...interfaceFields,
+      hashAttribute: rule.hashAttribute,
+      status: rule.status,
+      // TODO are these mandatory
+      datasourceId: interfaceFields.datasourceId,
+      exposureQueryId: interfaceFields.exposureQueryId,
+      autoSnapshots: true,
+      guardrailMetricIds: interfaceFields.guardrailMetricIds,
+      maxDurationDays: interfaceFields.maxDurationDays,
+      featureId: feature.id,
+      ruleId: rule.id,
+      environment: environment,
+    };
+
+    const safeRollout = await context.models.safeRollout.create(
+      safeRolloutCreateProps
+    );
+    if (safeRollout && safeRollout.id) rule.safeRolloutId = safeRollout.id;
+  }
   const revision = await getDraftRevision(context, feature, parseInt(version));
-  const resetReview = resetReviewOnChange({
-    feature,
-    changedEnvironments: [environment],
-    defaultValueChanged: false,
-    settings: org?.settings,
-  });
-  await addFeatureRule(
-    revision,
-    environment,
-    rule,
-    res.locals.eventAudit,
-    resetReview
-  );
+  if (
+    (rule.type === "safe-rollout" && rule.safeRolloutId) || // make sure the safe rollout is created
+    rule.type !== "safe-rollout"
+  ) {
```

Comment:
Instead of wrapping this in an `if` block, just throw above if it fails to create.